### PR TITLE
Extract Aspire.Hosting.Kafka.Tests project

### PR DESCRIPTION
**Pull request by eerhardt**
Contributes to #3185
Contributes to #4294

cc @Alirexaa
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4910)

### CODE REVIEW

File: Aspire.sln [hunk 1]
```diff
@@ -505,11 +505,16 @@ EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Components", "Components", "{C424395C-1235-41A4-BF55-07880A04368C}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Hosting", "Hosting", "{830A89EC-4029-4753-B25A-068BAE37DEC7}"
+	ProjectSection(SolutionItems) = preProject
+		tests\Aspire.Hosting.Kafka.Tests\Aspire.Hosting.Kafka.Tests.csproj = tests\Aspire.Hosting.Kafka.Tests\Aspire.Hosting.Kafka.Tests.csproj
+	EndProjectSection
```
Comment (eerhardt) [inline Aspire.sln:510]: Hmm, not sure what this is. Will revert.

File: tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs [hunk 2]
```diff
@@ -306,25 +306,6 @@ public void EnsureAllRabbitMQManifestTypesHaveVersion0Suffix()
         Assert.Equal("container.v0", server.GetProperty("type").GetString());
     }
 
-    [Fact]
-    public void EnsureAllKafkaManifestTypesHaveVersion0Suffix()
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs:310]: I don't think this test adds value over the `AddKafkaTests.VerifyManifest` test. That test verifies it is a container.v0 resource.

Comment (radical) [inline tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs:310]: Why are we dropping this test?

Comment (radical) [reply tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs:310]: ok :)

**Review by sebastienros — APPROVED** [review]
Only thing that I notice is that you removed it from E2E tests, which is fine by me.

**Pull request merged**